### DISCUSSION TITLE
Foreign key references to `User` table

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [neonexus/sails-react-bootstrap-webpack#103] [Q&A] [answered]
Hi @neonexus 

I have a table (says `AppDocs`) in which all the records have to have the user who updated it for audiiting and other purposes.

If I reference the attribute `updatedBy` with `model: 'user'`, it would not be feasible when that particular user gets permanently deleted from `User` table.

I am thinking to use the `email` or `username` of the user instead of a foreign key referencing `User` in `AppDocs`. This way the `updatedBy` won't get affected in case of permanent deletions in `User` table.

Do you have any other better suggestions regarding this? Thank you.

**Comment by neonexus** [comment, marked as answer]
@MaheshkumarSundaram This is why it is best practice to never delete a user record, but instead, use soft-deletes, like I do [in the user model](https://github.com/neonexus/sails-react-bootstrap-webpack/blob/release/api/models/User.js#L101). I have a `deletedAt` (and `deletedBy`) which I use in [the login controller](https://github.com/neonexus/sails-react-bootstrap-webpack/blob/release/api/controllers/common/login.js#L52); but now that I look at it, needs to be added to the `isLoggedIn` policy... So I have a new update coming soon...

Essentially, the idea is, if even you recreate the user, you can still use the deleted user as a pointer, not just to the user, but that user in time.

**Reply by neonexus** [reply]
That said, you really should never delete data, unless it is absolutely necessary. It just cause problems. Soft-deletes are your friends; ESPECIALLY if you have auditing reasons!

**Reply by MaheshkumarSundaram** [reply]
Thank you @neonexus
